File: src/dataset/Java/resistor_color_trio_Exercism_java_tusktenon/gemini/GeminiFlash_ResistorColorTrio_v1.java
class ResistorColorTrio {
  private static final int[] COLOR_VALUES = {0, 1, 2, 3, 4, 5, 6, 7, 8, 9};
  private static final String[] COLOR_NAMES = {
    "black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white"
  };

  String label(String[] colors) {
    long r = resistance(colors);
    if (r < 1000) return r + " ohms";
    if (r < 1_000_000) return r / 1000 + " kiloohms";
    if (r < 1_000_000_000) return r / 1_000_000 + " megaohms";
    return r / 1_000_000_000 + " gigaohms";
  }

  private long resistance(String[] colors) {
    int firstDigit = colorValue(colors[0]);
    int secondDigit = colorValue(colors[1]);
    int multiplier = colorValue(colors[2]);

    long r = (long) (firstDigit * 10 + secondDigit);
    if (multiplier < 10) {
        for (int i = 0; i < multiplier; i++) {
            r *= 10;
        }
    } else {
        r = 0;
    }
    return r;
  }

  private int colorValue(String color) {
    for (int i = 0; i < COLOR_NAMES.length; i++) {
      if (COLOR_NAMES[i].equals(color)) {
        return COLOR_VALUES[i];
      }
    }
    return -1; // Handle invalid color (though the problem doesn't specify this)
  }
}

File: src/dataset/typescript/resistor-color-trio_Exercism-typescript-FilipeCerejo/gemini/GeminiFlash_resistor-color-trio_v1.ts
const CV = {
  black: 0,
  brown: 1,
  red: 2,
  orange: 3,
  yellow: 4,
  green: 5,
  blue: 6,
  violet: 7,
  grey: 8,
  white: 9
}

type Color = keyof typeof CV;

export function decodedResistorValue(colors: Color[]) {
  const digit1 = CV[colors[0]];
  const digit2 = CV[colors[1]];
  const multiplier = CV[colors[2]];

  const rawResistorValue = (digit1 * 10 + digit2) * Math.pow(10, multiplier);

  if (rawResistorValue >= 1000000) {
    return `${rawResistorValue / 1000000} megaohms`;
  } else if (rawResistorValue >= 1000) {
    return `${rawResistorValue / 1000} kiloohms`;
  } else {
    return `${rawResistorValue} ohms`;
  }
}